README
MooseX-Storage version 0.32

INSTALLATION

To install this module, run the following commands:

    perl Makefile.PL
    make
    make test
    make install

Alternatively, to install with Module::Build, you can use the following commands:

    perl Build.PL
    ./Build
    ./Build test
    ./Build install

DEPENDENCIES

    Moose
    String::RewritePrefix
    Test::More
    Test::Deep
    Test::Fatal
    Test::Requires
    
OPTIONAL DEPENDENCIES

    JSON::Any
    IO::File
    IO::AtomicFile
    Test::YAML::Valid
    Test::Deep::JSON
    Digest
    Digest::SHA1
    Data::Dumper

COPYRIGHT AND LICENCE

Copyright (C) 2007-2008 Infinity Interactive

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.



Changes
Revision history for MooseX-Storage

{{$NEXT}}

0.53      2020-04-18 03:46:55Z
  - minor prerequisite improvements

0.52      2016-06-22 03:22:12Z
  - when looking for a serializer for a type constraint, always use the
    user-defined type handler first, before recursing into parent types

0.51      2016-02-10 20:06:41Z (TRIAL RELEASE)
  - support serialization of "maybe" types created by MooseX::Types (PR#12)

0.50      2015-05-05 20:09:42Z
  - revert changes in 0.49 for deeply-nested objects (see RT#104106)

0.49      2015-03-29 00:24:43Z
  - now handling more deeply-nested objects (PR#7, RT#81236, thanks Dmitry
    Latin!)
  - fix error in DateTime example (PR#8, thanks sillitoe!)

0.48      2014-10-12 02:01:31Z
  - switched the JSON serializer from JSON::Any to JSON::MaybeXS
  - interactive prompting at install time is now back!

0.47      2014-08-23 22:34:21Z
  - report the versions of all JSON and YAML backends during tests, in an
    attempt to track down an elusive bug
  - now properly serializing attributes that are set to the undefined value
    (github #6, thanks, Jason Pope!)
  - fix compatibility problem with MooseX::Role::Parameterized 1.03+ and
    MooseX::Storage::Format::JSONpm (some internal details changed that we
    were relying on)

0.46      2014-03-23 18:26:44Z
  - numerous small documentation clarifications and fixes

0.45      2013-12-22 00:48:10Z
  - new 'Value' type handler added (thanks David Golden)

0.44      2013-11-30 20:09:54Z
  - re-release to fix compile test

0.43      2013-09-11 01:47:40Z
  - removed use of deprecated Class::MOP::load_class

0.42      2013-09-08 18:00:22Z
  - all features are now configured as x_default => 1 in metadata (not yet
    supported by any cpan clients), signalling to automatically add to prereqs
    when installing non-interactively
  - re-release with properly encoded META.yml

0.41      2013-09-03 01:05:15Z
  - re-release with updated Dist::Zilla plugins to clear the dynamic_config
    flag in metadata
  - repository has moved to the GitHub Moose organization

0.40      2013-08-29 01:57:00Z
  - replaced last use of Best (held over since 2010!) to YAML::Any
  - fixed another test making use of optional features

0.39      2013-07-27 00:02:26Z
  - re-release as stable (full working Dist::Zilla conversion)

0.38      2013-07-20 16:42:03Z (TRIAL RELEASE)
  - re-release with new compile test, which properly declares all its prerequisites

0.37      2013-07-20 00:41:17Z (TRIAL RELEASE)
  - fixed compile tests so that modules using optional features are skipped
    (from 0.36-TRIAL)

0.36      2013-07-17 04:41:39Z (TRIAL RELEASE)
  - convert to Dist::Zilla, with new optional prereq handling

0.35    2013-07-15
  - properly skip optional tests when the required JSON backends aren't
    available

0.34    2013-06-20
  - JSON storage works when Cpanel::JSON::XS is the only JSON backend installed

0.33    2013-03-29
  - always JSONify data with sorted keys, to handle hash order randomization
    introduced in 5.17.* (RT#84287, Karen Etheridge)

0.32    2012-12-04
  - Change to use core Digest and Digest::SHA dists, rather than Digest::SHA1.
    RT#69811
  - Create a JSON::Any object, so setting $ENV{JSON_ANY_CONFIG} works.
    Adjust tests that were setting the wrong values and the default
    happened to work.
  - Don't rely on hash key order in tests (RT#81695)

0.31    2012-02-28
  - Add example for add_custom_type_handler to the MooseX::Storage::Engine docs. (perigrin)

  - Incorrect documentation for ->unpack method's inject parameter fixed.

  - The test suite now uses Test::Fatal instead of Test::Exception (Karen
    Etheridge).

0.30    2011-04-29
  - Make Makefile.PL features work again. RT#67170

0.29    2010-11-17
  - Remove use of Test::TempDir in favor of File::Temp (gphat)

0.28    2010-10-03
  - Make the engine_traits parameter be able to use a '+My::Trait::Name'
    prefix to specify a trait name which isn't in the
    MooseX::Storage::Engine::Trait namespace
  - Misc POD and test cleanups.

0.27    2010-04-15
  - use of parameterized roles (first added in 0.22) now works in Deferred
    Example: $object->freeze({ format => [ $role_name => \%args ] })

0.26    2010-03-25
  - Fix URI for repository in metadata.
  - Fix infinite recursion when collapsing objects which overload stringify
    to freeze themselves.

0.25    2010-03-09
  - Add support for Union types (bumps Moose dep to 0.99)

0.24    2010-01-11
  - Add additional YAML implementation tests.
  - Switch from using Best to YAML::Any.
  - Remove Test::YAML::Valid.

0.23    2009-11-10
  - Revert whitespace change in tests which breaks the tests if you
    have Test::YAML::Valid installed.

0.22    2009-11-05
  - Fix warnings when types do not have a parent type.
  - allow the use of roles outside MooseX::Storage:: (rjbs)
  - allow the use of parameterized roles (rjbs)

0.21    2009-07-14
  - Fix inconsistent dist versions with Perl::Version

0.20    2009-07-14
  - Add support for Storage( traits => [...] ) with a MooseX::Storage::Traits::XX
    namespace to alter the behaviour of the storage engine.
    (Suggestions for how to make this less fugly appreciated)
  - Add feature to disable cycle checking, either via engine trait or option
    - add docs & tests (including 1 TODO test)
  - Add feature to only serialize attributes whose predicate returns 'true'
  - Implement the unpack( $data, inject => {...} ) feature.
    - add docs & tests
  - Move Base::WithChecksum to build upon Basic.pm, avoid code duplication
  - Add extra test to ensure restoring fails if required argument isn't provided
  - Fix a bug where if a required constructor argument was not serialized, it was
    impossible to ->unpack again.

0.19    2009-07-14
  - Skip pod tests when not in author mode

0.18    2009-06-02
  - Fix test failures without JSON::Any. Patch from RT#46343 (Nicholas Clarke)
  - Added support for Maybe[...] constraints fixing RT#43165 (plu)

0.17    2009-04-06

Build.PL

use strict;
use warnings;

use Module::Build;

my $build = Module::Build->new(
    module_name => 'MooseX::Storage',
    license => 'perl',
    requires => {   
        'Moose'        => '0.20',
        # you should have at least one 
        # serialization format
        'JSON::Any'    => '0',
        'Best'         => '0', # << this if for loading YAML
        'Storable'     => '0',
        # and the ability to save the 
        # file to disk        
        'IO::File'     => '0',    
    },
    optional => {
        'IO::AtomicFile'    => '0',  
        # these are only used in the 
        # test, I recommend them, but 
        # there is no need to force 
        # them on people :)
        'Test::YAML::Valid' => '0',
        'Test::JSON'        => '0',        
        # this if for the basic role with checksum
        'Digest'            => '0', 
        'Digest::SHA1'      => '0',    
        'Data::Dumper'      => '0',        
    },
    build_requires => {
        'Test::More'       => '0.62',
        'Test::Exception'  => '0.21',
        'Test::Deep'       => '0',        
    },
    create_makefile_pl => 'traditional',
    recursive_test_files => 1,
    add_to_cleanup => [
        'META.yml', '*.bak', '*.gz', 'Makefile.PL',
    ],
);

$build->create_build_script;

